app/IATI/Traits/FillDefaultValuesTrait.php

```diff
@@ -0,0 +1,246 @@
+     * @param $datum
+     *
+     * @return void
+     */
+    public function setTempAmount(string $key, $datum): void
+    {
+        if ($key === 'amount') {
+            $this->tempAmount = $datum;
+        }
+    }
+
+    /**
+     * Sets default language if language is empty && non-empty narrative['narrative'].
+     *
+     * @param array $data
+     * @param string $key
+     * @param $datum
+     * @param $defaultValues
+     *
+     * @return void
+     */
+    public function setLanguage(array &$data, string $key, $datum, $defaultValues): void
+    {
+        if ($key === 'language' && empty($datum) && !empty($this->tempNarrative)) {
+            $data['language'] = Arr::get($defaultValues, 'default_language', null);
+        }
+    }
+
+    /**
+     * Sets default currency if currency is empty && non-empty amount['amount'].
+     *
+     * @param array $data
+     * @param string $key
+     * @param $datum
+     * @param $defaultValues
+     *
+     * @return void
+     */
+    public function setCurrency(array &$data, string $key, $datum, $defaultValues): void
+    {
+        if ($key === 'currency' && empty($datum) && !empty($this->tempAmount)) {
+            $data['currency'] = Arr::get($defaultValues, 'default_currency', null);
+        }
+    }
+
+    /**
+     * Overriding base Repository class's store method.
+     * Modified to populate default field values on save.
+     *
+     * @param array $data
+     *
+     * @inheritDoc
+     *
+     * @return Model
+     */
+    public function store(array $data): Model
+    {
+        $defaultFieldValues = $this->resolveDefaultValues($data);
+        $data = $this->populateDefaultFields($data, $defaultFieldValues);
+        $data['default_field_values'] = $defaultFieldValues;
+
+        return $this->model->create($data);
+    }
+
+    /**
+     * Overriding base Repository class's update method.
+     * Modified to populate default field values on update.
+     *
+     * @param $id
+     * @param $data
+     *
+     * @inheritDoc
+     *
+     * @return bool
+     */
+    public function update($id, $data): bool
+    {
+        $defaultFieldValues = $this->resolveDefaultValues($data);
+        $data = $this->populateDefaultFields($data, $defaultFieldValues);
+        $data['default_field_values'] = $defaultFieldValues;
+
+        return $this->model->find($id)->update($data);
+    }
+
+    /**
+     * Set Default values for the imported csv activities.
+     *
+     * @param string $id
+     * @param $data
+     *
+     * @return array
+     */
+    protected function resolveDefaultValues($data): array
+    {
+        $defaultValueTemplate = [
+            'default_currency'    => '',
+            'default_language'    => '',
+            'hierarchy'           => '',
+            'budget_not_provided' => '',
+            'humanitarian'        => '',
+        ];
+
+        $defaultValuesFromImport = isset($data['default_field_values']) && !empty($data['default_field_values'])
```

Comment:
use a variable to represent $data['default_field_values'] instead of accessing it multiple time